src/span.ts

```diff
@@ -1,22 +1,52 @@
 import { span } from "./extension_wrapper"
 import { Data } from "./internal/data"
 
+export type SpanData = {
+  closed?: boolean
+  name?: string
+  namespace?: string
+  parent_span_id?: string
+  span_id?: string
+  start_time_seconds?: number
+  start_time_nanoseconds?: number
+  trace_id?: string
+  error?: { name: string; message: string; backtrace: Array<string> }
+  attributes?: { [key: string]: string }
+}
 export class Span {
-  ref: unknown
+  #ref: unknown
 
   constructor(ref: unknown) {
-    this.ref = ref
+    this.#ref = ref
   }
 
   public close(): this {
-    span.closeSpan(this.ref)
+    span.closeSpan(this.#ref)
     return this
   }
 
   public setError(name: string, message: string, stackdata: string): this {
     const parsedStackdata = Data.generate(stackdata.split("\n"))
 
-    span.addSpanError(this.ref, name, message, parsedStackdata)
+    span.addSpanError(this.#ref, name, message, parsedStackdata)
     return this
   }
+
+  /**
+   * Returns a SpanData object representing the internal Span in the extension.
+   *
+   * @private
+   */
+  public toObject(): SpanData {
+    const json = span.spanToJSON(this.#ref)
+
+    // if this is true, then the span has been garbage collected
+    // @TODO: i feel that this could have better ergonomics on the agent
+    // side. come up with something better here later.
```

Comment:
```suggestion
    // If the span JSON is empty, the span has been closed.
```

I know it's an old comment, but let's update it.
We're not going to change it returning an empty string any time soon I think.